{"name": "test_003", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: './temp/temp_pic.png'", "trace": "self = <TestCase.test_admin.TestAdmin object at 0x102135430>\ndrivers = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"1c25be16f83aa5c63235d7ca7e6c3b18\")>\n\n    @allure.story('管理员重置租户密码用例')\n    def test_003(self, drivers):\n        reset = Admin(drivers)\n        with allure.step('点击重置密码按钮'):\n            reset.click_reset_button()\n        with allure.step('填入相关信息'):\n            reset.input_reset_adminpassword(ini.password)\n            reset.input_reset_newpassword(ini.newpassword)\n            reset.input_reset_confirmpassword(ini.newpassword)\n            reset.input_reset_verify_code(ini.verify_code)\n        with allure.step('点击确定'):\n            reset.click_reset_sure()\n        with allure.step('判断是否重置成功'):\n            text = reset.get_reset_notice()\n            log.info(text)\n            assert text == \"密码重置成功！\"\n>           save_pic(drivers)\n\ntest_admin.py:77: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../utils/picture_save.py:7: in save_pic\n    allure.attach.file(\"./temp/temp_pic.png\", attachment_type=allure.attachment_type.PNG)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py:203: in file\n    plugin_manager.hook.attach_file(source=source, name=name, attachment_type=attachment_type, extension=extension)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_hooks.py:513: in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_manager.py:120: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_pytest/listener.py:256: in attach_file\n    self.allure_logger.attach_file(uuid4(), source, name=name, attachment_type=attachment_type, extension=extension)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/reporter.py:160: in attach_file\n    plugin_manager.hook.report_attached_file(source=source, file_name=file_name)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_hooks.py:513: in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_manager.py:120: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/logger.py:39: in report_attached_file\n    shutil.copy2(source, destination)\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py:475: in copy2\n    copyfile(src, dst, follow_symlinks=follow_symlinks)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nsrc = './temp/temp_pic.png'\ndst = PosixPath('/Users/a1234/dsmp-selenium+pytest+allure/TestCase/report/result/68e3fce8-eb32-4e3a-a0bd-18d8fc818643-attachment.png')\n\n    def copyfile(src, dst, *, follow_symlinks=True):\n        \"\"\"Copy data from src to dst in the most efficient way possible.\n    \n        If follow_symlinks is not set and src is a symbolic link, a new\n        symlink will be created instead of copying the file it points to.\n    \n        \"\"\"\n        sys.audit(\"shutil.copyfile\", src, dst)\n    \n        if _samefile(src, dst):\n            raise SameFileError(\"{!r} and {!r} are the same file\".format(src, dst))\n    \n        file_size = 0\n        for i, fn in enumerate([src, dst]):\n            try:\n                st = _stat(fn)\n            except OSError:\n                # File most likely does not exist\n                pass\n            else:\n                # XXX What about other special files? (sockets, devices...)\n                if stat.S_ISFIFO(st.st_mode):\n                    fn = fn.path if isinstance(fn, os.DirEntry) else fn\n                    raise SpecialFileError(\"`%s` is a named pipe\" % fn)\n                if _WINDOWS and i == 0:\n                    file_size = st.st_size\n    \n        if not follow_symlinks and _islink(src):\n            os.symlink(os.readlink(src), dst)\n        else:\n>           with open(src, 'rb') as fsrc:\nE           FileNotFoundError: [Errno 2] No such file or directory: './temp/temp_pic.png'\n\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py:260: FileNotFoundError"}, "steps": [{"name": "点击重置密码按钮", "status": "passed", "start": 1722925300079, "stop": 1722925301149}, {"name": "填入相关信息", "status": "passed", "start": 1722925301150, "stop": 1722925303415}, {"name": "点击确定", "status": "passed", "start": 1722925303415, "stop": 1722925304451}, {"name": "判断是否重置成功", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: './temp/temp_pic.png'\n", "trace": "  File \"/Users/a1234/dsmp-selenium+pytest+allure/TestCase/test_admin.py\", line 77, in test_003\n    save_pic(drivers)\n  File \"/Users/a1234/dsmp-selenium+pytest+allure/utils/picture_save.py\", line 7, in save_pic\n    allure.attach.file(\"./temp/temp_pic.png\", attachment_type=allure.attachment_type.PNG)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py\", line 203, in file\n    plugin_manager.hook.attach_file(source=source, name=name, attachment_type=attachment_type, extension=extension)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    raise exception.with_traceback(exception.__traceback__)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_pytest/listener.py\", line 256, in attach_file\n    self.allure_logger.attach_file(uuid4(), source, name=name, attachment_type=attachment_type, extension=extension)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/reporter.py\", line 160, in attach_file\n    plugin_manager.hook.report_attached_file(source=source, file_name=file_name)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    raise exception.with_traceback(exception.__traceback__)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/logger.py\", line 39, in report_attached_file\n    shutil.copy2(source, destination)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py\", line 475, in copy2\n    copyfile(src, dst, follow_symlinks=follow_symlinks)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py\", line 260, in copyfile\n    with open(src, 'rb') as fsrc:\n         ^^^^^^^^^^^^^^^\n"}, "attachments": [{"source": "68e3fce8-eb32-4e3a-a0bd-18d8fc818643-attachment.png", "type": "image/png"}], "start": 1722925304451, "stop": 1722925304596}], "attachments": [{"name": "log", "source": "ef44bf63-3c19-492b-9362-c384bb10c2cc-attachment.txt", "type": "text/plain"}, {"name": "stderr", "source": "e769c7ee-11f3-4d9b-ae03-3ae22d9e7b8a-attachment.txt", "type": "text/plain"}], "start": 1722925300079, "stop": 1722925304599, "uuid": "94fe2ce1-57fd-4c40-adb8-e10e362cfbee", "historyId": "cf7ca284712f043645afd9a1b8931028", "testCaseId": "cf7ca284712f043645afd9a1b8931028", "fullName": "TestCase.test_admin.TestAdmin#test_003", "labels": [{"name": "story", "value": "管理员重置租户密码用例"}, {"name": "feature", "value": "测试超管功能"}, {"name": "parentSuite", "value": "TestCase"}, {"name": "suite", "value": "test_admin"}, {"name": "subSuite", "value": "TestAdmin"}, {"name": "host", "value": "1234s-MacBook-Air.local"}, {"name": "thread", "value": "94539-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "TestCase.test_admin"}]}